bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/Utils.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mideaac.internal;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.Random;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.jose4j.base64url.Base64;
+import org.openhab.core.util.HexUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+
+/**
+ * The {@link Utils} class defines common byte and String array methods
+ * which are used across the whole binding.
+ *
+ * @author Jacek Dobrowolski - Initial contribution
+ * @author Bob Eckhoff - JavaDoc, reversed array and refined query String method
+ */
+@NonNullByDefault
+public class Utils {
+    private static Logger logger = LoggerFactory.getLogger(Utils.class);
+
+    static byte[] empty = new byte[0];
+
+    /**
+     * Converts byte array to upper case hex string
+     * 
+     * @param bytes bytes to convert
+     * @return string of upper case hex chars
+     */
+    public static String bytesToHex(byte[] bytes) {
```

Comment:
The content of this method was adapted according to https://github.com/openhab/openhab-addons/pull/17749/files#r1961996508

Now it is just a proxy and everywhere it is called it should be replaced by the core variant directly.

Reply:
I don't get the reference. It clicks back my PR 17749

Reply:
The link is referring to thisPR and it shows the file it is about, you might need to click 'load file' if it is to large to show the comment.
Anyway, it looks like all these occurences have been adapted in the mean time as i sugested. thanks